### PR TITLE
feat(STONEINTG-399): move cache functions to separate package

### PR DESCRIPTION
**Pull request by dirgim**
* Update controllers to use cache package
* Update unit tests to use cache package
* Rename binding index functions to reflect usage

Signed-off-by: dirgim <kpavic@redhat.com>

### CODE REVIEW

File: cache/cache.go [hunk 1]
```diff
@@ -44,8 +45,8 @@ func SetupReleaseCache(mgr ctrl.Manager) error {
 		"spec.snapshot", releaseIndexFunc)
 }
 
-// SetupApplicationCache adds a new index field to be able to search SnapshotEnvironmentBindings by Environment.
-func SetupApplicationCache(mgr ctrl.Manager) error {
+// SetupBindingEnvironmentCache adds a new index field to be able to search SnapshotEnvironmentBindings by Environment.
+func SetupBindingEnvironmentCache(mgr ctrl.Manager) error {
 	applicationIndexFunc := func(obj client.Object) []string {
```
Comment (hongweiliu17) [inline cache/cache.go:50]: Is it better to change variable name `applicationIndexFunc` to `bindingIndexFunc`? But this is not related to this function movement.

**Review by hongweiliu17 — APPROVED** [review]
LGTM.

File: cache/cache.go (hunk 1 above)
Comment (dirgim) [reply cache/cache.go]: Good idea, renamed to `bindingEnvironmentIndexFunc` for more clarity, thanks.

**Review by MartinBasti — APPROVED** [review]
